src/coin/skycoin/models/testsutil.go

```diff
@@ -0,0 +1,483 @@
+	if txn == nil {
+		skyTxn, err := makeTransaction(t)
+		require.NoError(t, err)
+		txn = &skyTxn
+	}
+	uitxn, err := NewUninjectedTransaction(txn, fee)
+	require.NoError(t, err)
+	return uitxn
+}
+
+func makeLocalWallet(t *testing.T) core.Wallet {
+	_, kd, err := makeUxOutWithSecret(t)
+	require.NoError(t, err)
+	require.NoError(t, err)
+	wallets := makeLocalWalletsFromKeyData(t, []KeyData{*kd})
+	wallet := wallets[0]
+	seed, seckeys, err2 := cipher.GenerateDeterministicKeyPairsSeed([]byte(kd.Mnemonic), kd.AddressIndex+1)
+	require.NoError(t, err2)
+	require.Equal(t, kd.SecKey, seckeys[kd.AddressIndex])
+	_, _, err3 := cipher.GenerateDeterministicKeyPair(seed)
+	require.NoError(t, err3)
+	wallet.GenAddresses(core.AccountAddress, uint32(kd.AddressIndex), 2, nil)
+	return wallet
+}
+
+var (
+	tmpWalletDir  string
+	testWalletEnv core.WalletEnv
+)
+
+func loadTestWalletEnv(t *testing.T) core.WalletEnv {
+	if tmpWalletDir == "" {
+		tmpDir, err := ioutil.TempDir("", testsuite.TestIDToken)
+		require.NoError(t, err)
+		tmpWalletDir = tmpDir
+	}
+	if testWalletEnv == nil {
+		testWalletEnv = NewWalletDirectory(tmpWalletDir)
+	}
+	return testWalletEnv
+}
+
+var whitespaceReplacer = strings.NewReplacer(" ", "-")
+
+func makeLocalWalletsFromKeyData(t *testing.T, keysData []KeyData) []core.Wallet {
+	walletsCache := make(map[string]core.Wallet)
+	wallets := make([]core.Wallet, len(keysData))
+	walletSet := loadTestWalletEnv(t).GetWalletSet()
+	for i, kd := range keysData {
+		walletID := whitespaceReplacer.Replace(kd.Mnemonic)
+		var w core.Wallet
+		var isFound bool
+		var err error
+		if w, isFound = walletsCache[kd.Mnemonic]; !isFound {
+			if w = walletSet.GetWallet(walletID); w == nil {
+				w, err = walletSet.CreateWallet(walletID, kd.Mnemonic, skywallet.WalletTypeDeterministic, false, func(string) (string, error) { return "", nil }, 0)
+				require.NoError(t, err)
+			}
+			walletsCache[kd.Mnemonic] = w
+		}
+		wallets[i] = w
+		w.GenAddresses(core.AccountAddress, 0, uint32(kd.AddressIndex+1), nil)
+		w.GenAddresses(core.ChangeAddress, 0, uint32(kd.AddressIndex+1), nil)
+	}
+	return wallets
+}
+
+type TransferOptions struct {
+	values map[string]interface{}
+}
+
+func (tOpt *TransferOptions) GetValue(key string) interface{} {
+	return tOpt.values[key]
+}
+
+func (tOpt *TransferOptions) SetValue(key string, value interface{}) {
+	tOpt.values[key] = value
+}
+
+func makeSpentOutput(uxout coin.UxOut, spentBkSeq uint64, spentTxId cipher.SHA256) (rOut readable.SpentOutput) {
+	rOut.Uxid = uxout.Hash().Hex()
+	rOut.Time = uxout.Head.Time
+	rOut.SrcBkSeq = uxout.Head.BkSeq
+	rOut.SrcTx = uxout.Body.SrcTransaction.Hex()
+	rOut.OwnerAddress = uxout.Body.Address.String()
+	rOut.Coins = uxout.Body.Coins
+	rOut.Hours = uxout.Body.Hours
+	rOut.SpentBlockSeq = spentBkSeq
+	rOut.SpentTxnID = spentTxId.Hex()
+	return
+}
+
+func randBytes(t *testing.T, n int) []byte {
```

Comment:
This is not specific to Skycoin altcoin plugin . Please move onto `util` package.